lib/exq_scheduler/schedule.ex

```diff
@@ -55,15 +57,20 @@ defmodule ExqScheduler.Schedule do
               enqueued_at: nil
 
     def decode(serialized) do
-      Poison.decode!(serialized, as: %__MODULE__{})
+      struct(__MODULE__, Serializer.decode!(serialized))
```

Comment:
This doesn't work in its current form because the keys returned by Serializer.decode! is of type binary and not atom. I thought just switching the key type to atom would do the trick, but we need to only use keys for top level, the args value should still use binary key

```
struct: %ExqScheduler.Schedule.Job{
  args: [%{scheduled_at: 1627533720.0}],
  class: "TestJob",
  enqueued_at: 1627533844,
  error_class: nil,
  error_message: nil,
  failed_at: nil,
  finished_at: nil,
  jid: "cfdf8a5d-ca4c-47d3-9529-35735bb205fe",
  processor: nil,
  queue: "default",
  retry: true,
  retry_count: 0
}
poison: %ExqScheduler.Schedule.Job{
  args: [%{"scheduled_at" => 1627533720.0}],
  class: "TestJob",
  enqueued_at: 1627533844,
  error_class: nil,
  error_message: nil,
  failed_at: nil,
  finished_at: nil,
  jid: "cfdf8a5d-ca4c-47d3-9529-35735bb205fe",
  processor: nil,
  queue: "default",
  retry: true,
  retry_count: 0
}
```

Reply:
Btw, I am also ok to just switch the implementation to Jason instead of keeping support for both jason/poison. I am assuming if we switch to jason, we can just `@derive Jason.Encoder`